### PR TITLE
Query: Upgrade version of Remotion.Linq

### LINKED ISSUE #4311: Query: InMemory: Incorrect QueryModel coming from ReLinq for groupjoin scenarios
Started failing after 94a2d12b

### PR DESCRIPTION
**Pull request by smitpatel**
Resolves #4311

### CODE REVIEW

File: build/dependencies.props [hunk 1]
```diff
@@ -8,7 +8,7 @@
     <JsonNetVersion>10.0.1</JsonNetVersion>
     <NETStandardImplicitPackageVersion>2.0.0-*</NETStandardImplicitPackageVersion>
     <NETStandardLibraryNETFrameworkVersion>2.0.0-*</NETStandardLibraryNETFrameworkVersion>
-    <RelinqVersion>2.1.1</RelinqVersion>
+    <RelinqVersion>2.2.0-alpha-002</RelinqVersion>
```
Comment (smitpatel) [inline build/dependencies.props]: @Eilon for 3rd party dependency change approval.

File: src/EFCore.Specification.Tests/Query/QueryTestBase.cs [hunk 2]
```diff
@@ -215,6 +215,21 @@ public abstract class QueryTestBase<TFixture> : IClassFixture<TFixture>
             => AssertQueryScalar<TItem1, TItem2, int>(actualQuery, expectedQuery, assertOrder);
 
         public virtual void AssertQueryScalar<TItem1, TItem2>(
+            Func<IQueryable<TItem1>, IQueryable<TItem2>, IQueryable<string>> query,
```
Comment (maumar) [inline src/EFCore.Specification.Tests/Query/QueryTestBase.cs]: this shouldn't be needed

Comment (smitpatel) [inline src/EFCore.Specification.Tests/Query/QueryTestBase.cs]: There is no overload which satisfy `IQueryable<string>`

File: build/dependencies.props (hunk 1 above)
Comment (Eilon) [inline build/dependencies.props]: Done!

**Pull request merged**